tests/bsim/bluetooth/host/att/sequential/dut/src/main.c

```diff
@@ -0,0 +1,368 @@
+}
+
+static void subscribed(struct bt_conn *conn,
+		       uint8_t err,
+		       struct bt_gatt_subscribe_params *params)
+{
+	ASSERT(!err, "Subscribe failed (err %d)\n", err);
+
+	ASSERT(params, "params is NULL\n");
+
+	SET_FLAG(is_subscribed);
+	/* spoiler: tester doesn't really have attributes */
+	LOG_INF("Subscribed to Tester attribute");
+}
+
+void subscribe(void)
+{
+	int err;
+
+	/* Handle values don't matter, as long as they match on the tester */
+	static struct bt_gatt_subscribe_params params = {
+		.notify = notified,
+		.subscribe = subscribed,
+		.value = BT_GATT_CCC_NOTIFY | BT_GATT_CCC_INDICATE,
+		.value_handle = HVX_HANDLE,
+		.ccc_handle = (HVX_HANDLE + 1),
+	};
+
+	err = bt_gatt_subscribe(dconn, &params);
+	ASSERT(!err, "Subscribe failed (err %d)\n", err);
+
+	WAIT_FOR_FLAG(is_subscribed);
+}
+
+static void send_write_handle(void)
+{
+	int err;
+	uint8_t data[2];
+	const struct bt_gatt_attr *attr = &test_gatt_service.attrs[2];
+
+	/* Inform tester which handle it should write to */
+	sys_put_le16(bt_gatt_attr_get_handle(attr), data);
+
+	err = bt_gatt_notify(dconn, attr, data, sizeof(data));
+	ASSERT(!err, "Failed to transmit handle for write (err %d)\n", err);
+}
+
+void test_procedure_0(void)
+{
+	LOG_DBG("Test start: ATT sequential protocol");
+	int err;
+
+	err = bt_enable(NULL);
+	ASSERT(err == 0, "Can't enable Bluetooth (err %d)\n", err);
+	LOG_DBG("Central: Bluetooth initialized.");
+
+	/* Test purpose:
+	 * Test Spec V.3 P.F 3.3.2 Sequential protocol
+	 *
+	 * Verify that a Zephyr host server/client combo can process
+	 * concurrently: one Request, one Indication, multiple
+	 * Notifications and multiple Commands.
+	 *
+	 * To do this, the application on the DUT will purposefully stall the
+	 * HCI TX thread, ensuring that the responses are not sent until the
+	 * tester has finished sending everything.
+	 *
+	 * Test procedure:
+	 *
+	 * [setup]
+	 * - connect ACL
+	 * - update data length (tinyhost doens't have recombination)
+	 * - dut: subscribe to INDICATE and NOTIFY on tester CHRC
+	 * - dut: send a handle the tester can write to
+	 *
+	 * [proc]
+	 * - tester: send one ATT write request
+	 * - tester: send one ATT indication
+	 * - tester: send two ATT notifications
+	 * - tester: send two ATT commands
+	 *
+	 * - dut: handle the REQuest, build & put the RSP PDU on the HCI TX queue
+	 * - dut: suspend the HCI TX thread
+	 * - dut: handle the INDication
+	 * - dut: handle the notifications
+	 * - dut: handle the (write) commands
+	 * - dut: resume the TX thread after a short while
+	 *
+	 * [verdict]
+	 * - all procedures complete successfully, no buffer allocation failures
+	 *   or timeouts.
+	 */
+	connect();
+	subscribe();
+
+	do_dlu();
+
+	send_write_handle();
+
+	WAIT_FOR_FLAG(one_indication);
+	WAIT_FOR_FLAG(two_notifications);
+	/* One REQ, two CMDs */
+	WAIT_FOR_FLAG(three_writes);
+
+	/* Send RSP to LL */
+	k_thread_resume(bt_testing_tx_tid_get());
+
+	k_sleep(K_SECONDS(30));
+
+	PASS("DUT done\n");
```

Comment:
Nicely done in terms of following the test

Reply:
thanks! I'll remove the leftover `k_sleep`